core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -102,6 +104,18 @@ public ClassicPluginStrategy(PluginManager pluginManager) {
         this.pluginManager = pluginManager;
     }
 
+    /**
+     * Sets whether the classloader should include stacktraces on {@link ClassNotFoundException}s.
+     *
+     * @param skipStacktraces {@link false} to skip stacktraces.
+     *        The classloader injects stacktraces by default.
+     * @since TODO
+     */
+    public ClassicPluginStrategy withOmitStacktracesOnClassNotFoundException(boolean skipStacktraces) {
```

Comment:
I do not think we need a public API for this. Stack traces should be omitted by default, and we can leave in a system property to assist with debugging in case of unexpected `ClassNotFoundException`s.

Reply:
I would prefer to have public API for Jenkinsfile Runner and a few other use-cases.
Fine to change the defaults if there is a consensus about it

Reply:
Maybe I do not understand. Why would you need a Java API if the default is what you want anyway?